src/openrct2/object/RideObject.cpp

```diff
@@ -212,134 +226,40 @@ void RideObject::Load()
             // 0x6DE946
             vehicleEntry->base_num_frames = CalculateNumVerticalFrames(vehicleEntry)
                 * CalculateNumHorizontalFrames(vehicleEntry);
-            vehicleEntry->base_image_id = cur_vehicle_images_offset;
-            uint32_t image_index = vehicleEntry->base_image_id;
+            uint32_t baseImageId = cur_vehicle_images_offset;
+            uint32_t imageIndex = baseImageId;
+            vehicleEntry->base_image_id = baseImageId;
 
-            if (vehicleEntry->PaintStyle != VEHICLE_VISUAL_RIVER_RAPIDS)
+            if (vehicleEntry->PaintStyle == VEHICLE_VISUAL_RIVER_RAPIDS)
             {
-                const auto numRotationFrames = vehicleEntry->NumRotationFrames();
-                uint32_t b = vehicleEntry->base_num_frames * numRotationFrames;
-
-                image_index += b;
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_GENTLE_SLOPES)
+                imageIndex += vehicleEntry->base_num_frames * 36;
```

Comment:
I think there's no point in having this for river rapids the paint code is almost identical to virginia reel and it doesn't have this.